ipfx/dataset/ephys_nwb_data.py

```diff
@@ -0,0 +1,388 @@
+import warnings
+import re
+
+import numpy as np
+import pandas as pd
+from dateutil import parser as dateparser
+
+from io import BytesIO
+import h5py
+from pynwb import NWBHDF5IO
+
+from pynwb.icephys import (
+    CurrentClampSeries, CurrentClampStimulusSeries,
+    VoltageClampSeries, VoltageClampStimulusSeries,
+    IZeroClampSeries)
+from ipfx.py2to3 import to_str
+
+from ipfx.stimulus import StimulusOntology
+from ipfx.dataset.ephys_data_interface import EphysDataInterface
+
+
+def get_scalar_value(dataset_from_nwb):
+    """
+    Some values in NWB are stored as scalar whereas others as np.ndarrays with dimension 1.
+    Use this function to retrieve the scalar value itself.
+    """
+
+    if isinstance(dataset_from_nwb, np.ndarray):
+        return dataset_from_nwb.item()
+
+    return dataset_from_nwb
+
+
+def get_finite_or_none(d, key):
+    try:
+        value = d[key]
+    except KeyError:
+        return None
+
+    if np.isnan(value):
+        return None
+
+    return value
+
+
+class EphysNWBData(EphysDataInterface):
+    """
+    Abstract base class for implementing an EphysDataInterface with an NWB file
+
+    Provides common NWB2 reading and writing functionality
```

Comment:
A couple of days a ago @NileGraddis and I were thinking that it makes sense to limit the scope of the EphysDataInterface to just reading capabilities to enable feature extraction. Thus, for a user who only cares about extracting features, it would only be necessary to implement the reading interface. In turn, Nile has implemented writing capabilities by directly accessing the pynwb.io object.